velox/exec/tests/TableWriteTest.cpp

```diff
@@ -610,3 +644,96 @@ TEST_F(TableWriteTest, writeNoFile) {
   execute(plan, std::make_shared<core::QueryCtx>(executor_.get()));
   ASSERT_TRUE(fs::is_empty(outputDirectory->path));
 }
+
+TEST_F(TableWriteTest, immutablePartitionsExistingTableError) {
+  auto input = makeVectors(rowType_, 10, 10);
+  auto outputDirectory = TempDirectoryPath::create();
+  auto plan = createInsertPlan(
+      input,
+      rowType_,
+      connector::hive::LocationHandle::TableType::kExisting,
+      outputDirectory->path);
+
+  VELOX_ASSERT_THROW(
+      AssertQueryBuilder(plan)
+          .connectorConfig(
+              kHiveConnectorId,
+              HiveConfig::kImmutablePartitions,
+              folly::to<std::string>("true"))
+          .copyResults(pool()),
+      "Unpartitioned Hive tables are immutable.");
+}
+
+TEST_F(TableWriteTest, immutablePartitionsNewTable) {
+  // When table type is NEW, we always return UpdateMode::kNew, no exception
```

Comment:
Please clarify that no exception should be thrown even if kImmutablePartitions is set to true and we are trying to insert rows into the table.

Reply:
Updated, thanks.